### PR TITLE
Add kafka to docker-compose and sha_generator script for local testing

### PR DESCRIPTION
**Pull request by epapbak**
- The `local_setup/sha_generator.go` allows to create a given number of SHA-256 digests and print them in the console
  - with `-produce`, the digests can be sent to the configured Kafka topic (see help for related arguments)
  - with `-store`, the digests can be stored in the 'image' table of the configured database as if previously populated by the Pyxis Gatherer.
- Adds Kafka and Zookeeper services to the `docker-compose.yml`, as well as an `init-kaka` container that creates a `test_sha` topic.

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [x] Output Encoding
- [x] Authentication and Password Management
- [x] Session Management
- [x] Access Control
- [x] Cryptographic Practices
- [x] Error Handling and Logging
- [x] Data Protection
- [x] Communication Security
- [x] System Configuration
- [x] Database Security
- [x] File Management
- [x] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: local_setup/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+- generate <NUM> SHAs and print on console - `go run sha_generator.go -num NUM`
+
+- specify which org ID and account - `go run sha_generator.go -org 12345 -account 12345678`
+  - This will create one SHA-256 as `-num` is not specified.
+  - `org` and `account` do not influence the result as it is not sent to Kafka nor stored in DB.
+
+- send generated data to Kafka (see help for related configuration) - `go run sha_generator.go -produce`
+- store  generated data in DB (see help for related configuration) - `go run sha_generator.go -store`
+
+- generate <NUM> SHAs and send to kafka - `go run sha_generator.go -num NUM -produce`
```
Comment (jdobes) [inline local_setup/README.md:15]: I was not able to connect to kafka in docker on port 9092 from localhost. Kafka seems to be verifying the hostname somehow and doesn't like I'm connecting to URL `localhost:9092` and not `kafka:9092`. After a bit googling I found a solution - setting up Kafka to have second listener for localhost (it has to be on different port), I had to set this in kafka.env:

```
KAFKA_CFG_LISTENER_SECURITY_PROTOCOL_MAP=PLAINTEXT:PLAINTEXT,PLAINTEXT_EXTERNAL:PLAINTEXT
KAFKA_CFG_LISTENERS=PLAINTEXT://:9092,PLAINTEXT_EXTERNAL://:9093
KAFKA_ADVERTISED_LISTENERS=PLAINTEXT://kafka:9092,PLAINTEXT_EXTERNAL://localhost:9093
```

After this and changing the port to 9093 in docker-compose.yml and sha_generator, `-produce` is working.

File: local_setup/sha_generator.go [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+		DBCONFIG.PGPassword,
+		DBCONFIG.PGHost,
+		DBCONFIG.PGPort,
+		DBCONFIG.PGDBName,
+		DBCONFIG.PGParams,
+	)
+	return
+}
+func prepareInsertDigestsStatement(shas []string) (statement string, statementArgs []interface{}) {
+	statement = `INSERT INTO image (digest, health_index) VALUES %s`
```
Comment (jdobes) [inline local_setup/sha_generator.go:180]: The health index column was removed from DB for now.

File: docker-compose.yml [hunk 3]
```diff
@@ -1,6 +1,44 @@
...
+    depends_on:
+      - kafka
+    entrypoint: [ '/bin/sh', '-c' ]
+    command: |
+      "
+      # blocks until kafka is reachable
+      kafka-topics.sh --bootstrap-server kafka:9092 --list
+
+      echo -e 'Creating kafka topics'
+      kafka-topics.sh --bootstrap-server kafka:9092 --create --if-not-exists --topic test_sha --replication-factor 1 --partitions 1
```
Comment (jdobes) [inline docker-compose.yml:36]: I'd probably use the topic name which will be used in real deployment instead of `test_sha`, but this is a minor thing.

**Review by jdobes — CHANGES_REQUESTED** [review]

File: local_setup/sha_generator.go (hunk 2 above)
Comment (epapbak) [reply local_setup/sha_generator.go:180]: I thought I updated that. I probably messed it up in between cherry picks

**Review by jdobes — APPROVED** [review]

**Pull request merged**